Doc/c-api/dict.rst

```diff
@@ -100,6 +100,11 @@ Dictionary Objects
    :meth:`__eq__` methods will get suppressed.
    To get error reporting use :c:func:`PyDict_GetItemWithError()` instead.
 
+   The caller must hold the :term:`GIL`.
+
+   .. versionchanged:: 3.10
+      The caller must now hold the :term:`GIL`.
```

Comment:
It looks bit curious that only this API has "The caller must hold the :term:`GIL` statement.

How about remove the first line and make `versionchanged` more descriptive?
E.g. "Calling this API without :term:`GIL` held had been allowed for historical reason.  It is not allowed now."

Reply:
Sure. I updated the doc. Does it look better?